keras_nlp/utils/preset_utils.py

```diff
@@ -155,6 +160,120 @@ def save_to_preset(
             metadata_file.write(json.dumps(metadata, indent=4))
 
 
+def _validate_tokenizer(preset, allow_incomplete=False):
+    config_path = get_file(preset, TOKENIZER_CONFIG_FILE)
+    if not os.path.exists(config_path):
+        if allow_incomplete:
+            logging.warning(
+                f"Tokenizer is missing from the preset directory {preset}."
+            )
+            return
+        else:
+            raise FileNotFoundError(
+                f"Tokenizer is missing from the preset directory {preset}."
+                "If you wish to upload model without a tokenizer, set `allow_incomplete=True`."
+            )
+    with open(config_path) as config_file:
+        config = json.load(config_file)
+    layer = keras.saving.deserialize_keras_object(config)
+
+    if not config["assets"]:
+        raise ValueError(
+            f"Tokenizer config file {config_path} is missing `asset`."
+        )
+
+    for asset in config["assets"]:
+        asset_path = os.path.join(preset, asset)
+        if not os.path.exists(asset_path):
+            raise FileNotFoundError(
+                f"Asset {asset} doesn't exist in the preset direcotry {preset}."
+            )
+    config_dir = os.path.dirname(config_path)
+    asset_dir = os.path.join(config_dir, TOKENIZER_ASSET_DIR)
+
+    tokenizer = get_tokenizer(layer)
+    if not tokenizer:
+        raise ValueError(f"Model or layer {layer} is missing tokenizer.")
+    tokenizer.load_assets(asset_dir)
+
+
+def _validate_backbone(preset):
+    config_path = get_file(preset, CONFIG_FILE)
+    with open(config_path) as config_file:
+        config = json.load(config_file)
+    # Check if backbone is deserializable.
+    keras.saving.deserialize_keras_object(config)
```

Comment:
I don't think we can do this. For large backbones, this will double the GPU footprint temporarily, and OOM when we are memory constrained.

Let's just skip for now. Maybe there are ways to use a `StatelessScope` to avoid state creation, but that seems tricky.

Reply:
Yeah! Makes sense! Thanks for pointing that out!